### PR TITLE
Add Text documentation to the site

### LINKED ISSUE #2222: Text Documentation

### PR DESCRIPTION
**Pull request by joshwooding**
closes #2222

### CODE REVIEW

File: site/docs/components/text/index.mdx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+title: Text
+data:
+  description: Text is a component that makes it easy to use our typography styles throughout a product or application, ensuring consistency and a unified look and feel. It includes attributes such as font size, weight, letter spacing, line height and decoration. Read more on our [Typography Foundation](/salt/foundations/typography) page.
```
Comment (feliciackh) [inline site/docs/components/text/index.mdx:4]: link doesn't work in preview

File: site/src/examples/text/Variant.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  Text,
+  H1,
+  H2,
+  H3,
+  H4,
+  Label,
+  Display1,
+  Display2,
+  Display3,
+  StackLayout,
```
Comment (libertymayc) [inline site/src/examples/text/Variant.tsx:12]: not using all of these?

File: site/src/examples/text/Styling.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  H1,
+  H2,
+  H3,
+  H4,
+  Label,
+  Display1,
+  Display2,
+  Display3,
+  StackLayout,
+} from "@salt-ds/core";
```
Comment (alycrys) [inline site/src/examples/text/Styling.tsx:13]:
```suggestion
import { Text } from "@salt-ds/core";
```

File: site/docs/components/text/examples.mdx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+<LivePreviewControls>
+
+<LivePreview componentName="text" exampleName="Styles">
+
+### Styles
+
+The Text component supports these sets of styles:
+
+- Display
+- Headings
```
[the remark this answers is not in the document]
Comment (joshwooding) [reply site/docs/components/text/examples.mdx:20]: This is taken from the foundations

File: site/src/examples/text/Styles.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+} from "@salt-ds/core";
+
+export const Styles = (): ReactElement => (
+  <StackLayout>
+    <H1>H1</H1>
+    <H2>H2</H2>
+    <H3>H3</H3>
+    <H4>H4</H4>
+    <Label>Label</Label>
+    <Display1>Display 1</Display1>
```
[the remark this answers is not in the document]
Comment (joshwooding) [reply site/src/examples/text/Styles.tsx:22]: I've asked Paul and updated accordingly

**Review by alycrys — APPROVED** [review]

**Review by sion-docs — APPROVED** [review]

**Pull request merged**